### PR TITLE
GH-11085 Migrate Team.GetAllTeamPageListing to Sync by default

### PR DESCRIPTION
**Pull request by lurcio**
#### Summary
Migrate "Team.GetAllTeamPageListing" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11085

### CODE REVIEW

File: store/sqlstore/team_store.go [hunk 1]
```diff
@@ -450,22 +450,19 @@ func (s SqlTeamStore) GetAllTeamListing() store.StoreChannel {
...
+	}
 
-		var data []*model.Team
-		if _, err := s.GetReplica().Select(&data, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
-			result.Err = model.NewAppError("SqlTeamStore.GetAllTeamListing", "store.sql_team.get_all_team_listing.app_error", nil, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	var data []*model.Team
+	if _, err := s.GetReplica().Select(&data, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
+		return nil, model.NewAppError("SqlTeamStore.GetAllTeamListing", "store.sql_team.get_all_team_listing.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (mounicapaladugu) [inline store/sqlstore/team_store.go]: Can you change this error message to GetAllTeamPageListing and run make i18n-extract to extract the strings from the source code to the file i18n/en.json? Also modify store.sql_team.get_all_team_listing.app_error --> store.sql_team.get_all_team_page_listing.app_error

Comment (lurcio) [reply store/sqlstore/team_store.go]: @mounicapaladugu Sure, will do. I'll take a look tomorrow, it's late here 😄

File: app/team.go [hunk 2]
```diff
@@ -649,11 +649,11 @@ func (a *App) GetAllPublicTeams() ([]*model.Team, *model.AppError) {
 }
 
 func (a *App) GetAllPublicTeamsPage(offset int, limit int) ([]*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetAllTeamPageListing(offset, limit)
```
Comment (jespino) [inline app/team.go]: You can just `return a.Srv.Store.Team().GetAllTeamPageListing(offset, limit)`

File: store/sqlstore/team_store.go [hunk 3]
```diff
@@ -450,22 +450,19 @@ func (s SqlTeamStore) GetAllTeamListing() store.StoreChannel {
...
+	if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		query = "SELECT * FROM Teams WHERE AllowOpenInvite = true ORDER BY DisplayName LIMIT :Limit OFFSET :Offset"
+	}
 
-		var data []*model.Team
-		if _, err := s.GetReplica().Select(&data, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
-			result.Err = model.NewAppError("SqlTeamStore.GetAllTeamListing", "store.sql_team.get_all_team_listing.app_error", nil, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	var data []*model.Team
```
Comment (jespino) [inline store/sqlstore/team_store.go]: nit: Probably we can rename here `data` to `teams` which is more meaningful.

File: store/storetest/team_store.go [hunk 4]
```diff
@@ -536,20 +536,17 @@ func testGetAllTeamPageListing(t *testing.T, ss store.Store) {
 	_, err = ss.Team().Save(&o4)
 	require.Nil(t, err)
 
-	if r1 := <-ss.Team().GetAllTeamPageListing(0, 10); r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		teams := r1.Data.([]*model.Team)
+	teams, err := ss.Team().GetAllTeamPageListing(0, 10)
+	require.Nil(t, err)
```
Comment (reflog) [inline store/storetest/team_store.go]: Nit: `require.NoError` is more appropriate here. 0/5

File: store/storetest/team_store.go [hunk 5]
```diff
@@ -561,36 +558,30 @@ func testGetAllTeamPageListing(t *testing.T, ss store.Store) {
 	_, err = ss.Team().Save(&o5)
 	require.Nil(t, err)
 
-	if r1 := <-ss.Team().GetAllTeamPageListing(0, 4); r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		teams := r1.Data.([]*model.Team)
+	teams, err = ss.Team().GetAllTeamPageListing(0, 4)
+	require.Nil(t, err)
```
Comment (reflog) [inline store/storetest/team_store.go]: Same nitpick as above

File: store/storetest/team_store.go (hunk 4 above)
Comment (jespino) [reply store/storetest/team_store.go]: No is not, it generates certain problems related to we don't return an error, we return a model.AppError, which is not exactly the same.

Comment (reflog) [reply store/storetest/team_store.go]: Ouch. Thanks for clarifying!

**Comment by lurcio** [comment]
@mounicapaladugu sorry for the delay in looking at this. I have incorporated your feedback. Running `make i18n-extract` has added more changes than I anticipated and I think this is breaking the build at the `Check i18n` step. Is this OK? Could you take a look please?

@jespino Thank you for your feedback, I have changed my PR as per your review.

**Comment by mounicapaladugu** [comment]
Hi @lurcio. Don't worry about the modification if it adds too much overhead. You can revert the error message to whatever it was before. Thanks!!

**Pull request merged**